pkg/volume/git_repo/git_repo.go

```diff
@@ -138,8 +140,15 @@ func (b *gitRepoVolumeBuilder) SetUpAt(dir string) error {
 		return err
 	}
 
-	if output, err := b.execCommand("git", []string{"clone", b.source}, dir); err != nil {
-		return fmt.Errorf("failed to exec 'git clone %s': %s: %v", b.source, output, err)
+	// If provided target directory
+	if "" != b.target {
+		if output, err := b.execCommand("git", []string{"clone", b.source, b.target}, dir); err != nil {
+			return fmt.Errorf("failed to exec 'git clone %s %s': %s: %v", b.source, b.target, output, err)
+		}
+	} else {
+		if output, err := b.execCommand("git", []string{"clone", b.source}, dir); err != nil {
+			return fmt.Errorf("failed to exec 'git clone %s': %s: %v", b.source, output, err)
+		}
 	}
```

Comment:
This will fail when clone directory is '.'

Actually, see my comments here: https://github.com/kubernetes/kubernetes/issues/15421#issuecomment-147963960

I'd like to update the code if we reached a agreement.